spec/unit/endpoint/authorization_spec.rb

```diff
@@ -50,6 +50,42 @@
       end
     end
 
+    describe 'evil hackers messing with redirection' do
+      before do
+        WebMock.stub_request(:post, "https://foobar.com/access_token_path")
+          .to_return(status: 200, body: 'access_token=token&token_type=bearer')
+
+        WebMock.stub_request(:get, "https://api.github.com/user?per_page=100")
+          .to_return(
+            status: 200,
+            body: JSON.dump(name: 'Piotr Sarnacki', login: 'drogus', gravatar_id: '123', id: 456, foo: 'bar'), headers: {'X-OAuth-Scopes' => 'repo, user, new_scope'}
+          )
+
+        cookie_jar['travis.state-github'] = state
+        Travis.redis.sadd('github:states', state)
+      end
+
```

Comment:
I would also add:
```
after do
  Travis.redis.srem('github:states', state)
end
```

just to avoid any side effect in other tests in the future

Reply:
Absolutely right. Thank you.